### PR TITLE
Enhancement: Change default k8s label selector to be from resource

### PR DESCRIPTION
**Pull request by graytonio**
## Proposed change

Enhance default k8s pod selector by using the api to automatically discover the service used by the ingress resource. Reduces need to manually set pod selectors and does not break existing functionality. I opened a discussion on this but I was unclear if I needed to upvotes for this since it's not really a new feature but please let me know if I do.

Closes #3348

## Type of change

- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation only
- [x] Other (please explain)

Enhancement of current functionality to integrate more natively in kubernetes.

## Checklist:

- [ ] If applicable, I have added corresponding documentation changes.
- [x] If applicable, I have reviewed the [feature](https://gethomepage.dev/latest/more/development/#new-feature-guidelines) and / or [service widget guidelines](https://gethomepage.dev/latest/more/development/#service-widget-guidelines).
- [x] I have checked that all code style checks pass using [pre-commit hooks](https://gethomepage.dev/latest/more/development/#code-formatting-with-pre-commit-hooks) and [linting checks](https://gethomepage.dev/latest/more/development/#code-linting).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] (Did not test multiple front end platforms due to it being a backend change)

### CODE REVIEW

**Comment by shamoon** [comment]
I dont use k8s so our k8s integration could probably use some love. My question about this is isnt it going to be a completely breaking change for existing users who aren't using an explicit podSelector?

**Comment by graytonio** [comment]
I'm working on that part right now I don't want it to be a breaking change.

The way I'm thinking about it right now if you have manually configured the app and namespace parameters in your config file either:
A. An ingress exists with that exact name and very likely that is the application you want to target
B. An ingress does not exist with that name and it will default to the current behavior

**Comment by graytonio** [comment]
I was able to validate that you can configure the services manually in the config file and it will still work as intended.

both working

File: src/utils/kubernetes/kubernetes-utils.js
Comment (shamoon) [inline src/utils/kubernetes/kubernetes-utils.js:1]: So does this mean existing users will get these errors?

**Comment by shamoon** [comment]
> I opened a discussion on this but I was unclear if I needed to upvotes for this since it's not really a new feature but please let me know if I do.
> 
> Closes #3348

Oh yea I just noticed this, the answer is yes. Also in general I'm going to be hesitant to make this change without getting confirmation from real k8s users. This setup has been around for a while and went through a few iterations (you can look through the history) to what seems to be one that makes folks happy, Im disinclined to suddenly change it without clear support & rationale.

If that does happen of course we can re-address this PR, but we dont leave PRs open in the meantime.

**Comment by graytonio** [comment]
Understandable if it gets the support we can reopen this.

**Pull request closed without merging**